test/lib/ufe/testGroupCmd.py

```diff
@@ -95,26 +100,25 @@ def testUsdGroup(self):
         childPaths = set([child.path() for child in parentChildrenPost])
 
         self.assertTrue(newGroupPath in childPaths)
-        self.assertTrue(ball25Path not in childPaths)
-        self.assertTrue(ball35Path not in childPaths)
+        self.assertTrue(ball5Path not in childPaths)
+        self.assertTrue(ball3Path not in childPaths)
 
         groupCmd.undo()
 
         parentChildrenUndo = parentHierarchy.children()
-        self.assertEqual(len(parentChildrenUndo), 35)
+        self.assertEqual(len(parentChildrenUndo), 6)
 
         childPathsUndo = set([child.path() for child in parentChildrenUndo])
         self.assertTrue(newGroupPath not in childPathsUndo)
-        self.assertTrue(ball25Path in childPathsUndo)
-        self.assertTrue(ball35Path in childPathsUndo)
+        self.assertTrue(ball5Path in childPathsUndo)
+        self.assertTrue(ball3Path in childPathsUndo)
 
         groupCmd.redo()
 
         parentChildrenRedo = parentHierarchy.children()
-        self.assertEqual(len(parentChildrenRedo), 34)
+        self.assertEqual(len(parentChildrenRedo), 5)
 
         childPathsRedo = set([child.path() for child in parentChildrenRedo])
         self.assertTrue(newGroupPath in childPathsRedo)
-        self.assertTrue(ball25Path not in childPathsRedo)
-        self.assertTrue(ball35Path not in childPathsRedo)
-
+        self.assertTrue(ball5Path not in childPathsRedo)
+        self.assertTrue(ball3Path not in childPathsRedo)
```

Comment:
Hey @fowlertADSK , prior to applying the restriction rules, one could use top_layer.ma scene for writing this test case. However, with the new rules in place, we can't really use it: 

The first error you would hit is this:

`Cannot reparent [Ball_25] defined on another layer. Please set [Assembly_room_set.usda] as the target layer to proceed`

And once you go to the correct target layer, you would hit this.

`Cannot reparent [Ball_25] with definitions or opinions on other layers. Opinions exist in [Ball.usd],[edits.usda],[Assembly_room_set.usda],[Ball.maya.usd],[Ball.shadingVariants.usda]`

Hence, creating a simpler version of ball_set scene and applying your previous group logics to it.